docs/containerization-design.rst

```diff
@@ -0,0 +1,358 @@
++-----------------------+--------------------------------------------------------------------------+
+| Azkaban Core          | Core Azkaban binaries/configs/secrets owned and managed by Azkaban       |
++-----------------------+--------------------------------------------------------------------------+
+| Azkaban JobTypes      | Binaries/Configs owned by JobType developers but are managed by Azkaban  |
+|                       | like KafkaPushJob, SparkJob, etc.                                        |
++-----------------------+--------------------------------------------------------------------------+
+
+* Azkaban Core forms the base docker image layered on top of a base image of choice: such as RHEL7.
+* Each of the above (Platform or jobtypes) will be packaged as a separate docker image layers on top of a simple
+  base image of choice. It is preferable to keep these images very small to optimize download times. Something
+  like a busybox/alpine image should suffice.
+* Individual images for job-types will allow independent development and release for the job-type developers without
+  any dependency on Azkaban. Here is an example image definition for KPJ (Kafka Push Job):
+
+.. code-block:: docker
+
+  FROM container-image-registry.corp.linkedin.com/rhel7-base-image/rhel7-base-image:0.16.9
+
+  ARG KPJ_URL=https://artifactory.corp.linkedin.com/artifactory/DDS/com/linkedin/kafka-push-job/kafka-push-job/0.2.61/kafka-push-job-0.2.61.jar
+
+  RUN curl $KPJ_URL --output ~/kafka-push-job-0.2.61.jar
+
+* There will be one init container [1]_ for each job-type using job-type docker images. This init container will move
+  the binaries and configs from the image to a mounted volume. The above specified volume will also be mounted for the
+  application container which will use Azkaban docker image.
+* Job-Type developers will use `Image Management APIs <#Image-Management-API>`_ for rolling out a job-type image.
+  The rolled out images, can then be used as a default version for that job-type. Flow writers will have the ability
+  to specify the version for the job-type image they want to use for their Azkaban flows via DSL.
+* A concept of version-set and version-number is introduced to uniquely identify the state of the dependencies
+  during the flow execution. This will also be useful during debugging the recreate exact environment for a flow
+  should there be any failures.
+
+Image Management API
+--------------------
+In order for the various image owners to give them their independent development life-cycle, we introduce various
+Admin API to:
+* Register image types along with their owners
+* Register new versions of already registered image types
+* Create/Update ramp-up plans for registered image versions
+
+For details regarding these API, please refer to :file:`docs/imageMgmtAPI.rst`
+
+
+Dispatch Logic
+--------------
+
+.. image:: docs/figures/dispatch_logic.png
```

Comment:
The box labeled `QueueProcessor Thread` seems unnecessary and can be merged with the `Poll for queued flows`. Consider removing.

Reply:
The box represents the start of the thread. If you still think it is un-necessary, I will remove it.